### PR TITLE
feat: create UI library module

### PR DESCRIPTION
**Pull request by AliYusuf95**
Solve #117

- [x] Create UI library module
- [x] Prefix resources used in UI
- [x] Use the module in the app

### CODE REVIEW

**Comment by or-else** [comment]
I would suggest a more unique name for the library, such as `co.tinode.tindroidui`. There is certainly no need for an additional namespace `android` in the name of the module (co.tinode._android_...).

**Comment by AliYusuf95** [comment]
That's fine @or-else
What about resources prefix should I use `tindroid_ui_` or `tindroidui_`.

**Comment by or-else** [comment]
How about naming everything `tinui`: `co.tinode.tinui` and `tinui_`? It's short, unique (no such modules exist at maven). What do you think?

**Comment by AliYusuf95** [comment]
I think it's good. 
By the way I prefer that the UI module naming should be similar to the SDK module; to be more convenient for the users. In this case the SDK module would be `tinsdk`?

**Comment by or-else** [comment]
`co.tinode.tinsdk`: yes, I like it.

File: app/src/main/java/co/tinode/tindroid/AboutDialogFragment.java [hunk 1]
```diff
@@ -26,7 +26,7 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
         }
 
         AlertDialog.Builder builder = new AlertDialog.Builder(activity);
-        View dialog = View.inflate(activity, R.layout.dialog_about, null);
+        View dialog = View.inflate(activity, R.layout.tindroid_dialog_about, null);
```
Comment (or-else) [inline app/src/main/java/co/tinode/tindroid/AboutDialogFragment.java:29]: There is no need for prefixes in the app. The prefixes should be used only in the libraries.

Comment (AliYusuf95) [reply app/src/main/java/co/tinode/tindroid/AboutDialogFragment.java:29]: That's right, I did this just to make the search/replace easier for me.
I'll remove the prefix once I have finish.

**Comment by AliYusuf95** [comment]
@or-else Currently a lot of features are tightly coupled with the application like (Contacts sync, Chats lists, Messages, etc.)
We need to have a list of features that should be in the UI module and the rest can be considered either plugins or as application customization.

Currently this is what my mind for UI:
- [ ] Chats lists
- [ ] Messages
- [ ] Images preview
- [ ] Attachments

**Comment by or-else** [comment]
I'm not concerned with the app being tightly coupled with the SDK and UI libraries. I do not foresee a case where I would want to swap this UI library for another in the app. So, the app is not of concern, the `tinui` is.

Let's take a step back: what is the intended use case for the `tinui`? I imagine that the use case for it is to be usable as components when building another chat app, like tindroid. If so, there is no clear line between the app and the UI library. The library can start small, as a set of a few clearly separable components. We can gradually move components from the app to the lib over time.

**Comment by AliYusuf95** [comment]
Exactly this is what I'm trying to achieve. 

What I understand that not every component will add to `tinui` will be used right away in the app. So it doesn't have to contain all features that used in the app. Ideally over time the app will rely on `tinui` components.
As you said this will give the freedom to gradually add more components to the lib.

**Comment by or-else** [comment]
Great, we agree.

I would start with creating components for message bubbles.

**Comment by or-else** [comment]
Then contact display.

**Comment by or-else** [comment]
This PR seems to be getting stale. Should I close it?

**Comment by AliYusuf95** [comment]
@or-else I was bit busy last month, but I made some progress creating some views and feed them from `Tinode` instance. I'll try to push finished parts by next week.

**Comment by or-else** [comment]
Thanks for the update.

**Comment by or-else** [comment]
Is this still relevant?

**Pull request closed without merging**